specs/switch-driver-capabilities.md

```diff
@@ -0,0 +1,48 @@
+
+# Expose network driver capabilites
+-----------
+
+[Issue 410](https://github.com/CCI-MOC/hil/issues/410)
+
+We want to expose switch driver capabilities. Capabilities include:
+
+- Jumbo frames are per-switch on our Dell switches and per-port on our Ciscos (#755)
+-  QoS (#384)
+- Disabling STP (#779)
+- DHCP/ARP Snooping (#471)
+- Enabling the tagging of DHCP packets with the port number (from @okrieg or @pjd-nu)
+
+# Solution
+-----------
+
+* Every switch driver will have a list called `capabilities` in the class. So it would look like
+`capabilities = ["require-native-network", "per-port-jumbo-frames"]`
+
+* and then there'll be a method  `get_capabilities()` that returns a boolean if a capability exists or not by checking it in the list.
+
+* an API that needs to test this, can call that method `switch.get_capabilities('require-native-network`) to ensure that the switch supports that method, and then proceed with the original request or error out.
+
+* `show_node()` can be updated to show the capabilities they support.
```

Comment:
Ideally we'd have a `show_switch` that would report this information. Might be worth considering whether we want to just add that first.

Reply:
True. `show_switch` should show this information. But I think we should show this information under `show_node()` just so that users can pick nodes with capabilities they require.